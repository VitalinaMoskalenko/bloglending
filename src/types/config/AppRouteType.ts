export enum AppRouteType {
  home = "/",
  postDetail = "/postDetail",
  postDetailsWithId = "/postDetail/:id",
}
