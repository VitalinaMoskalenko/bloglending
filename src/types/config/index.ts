export * from "./FontSizeType";
export * from "./ColorType";
export * from "./AppLocalsType";
export * from "./BreakpointType";
export * from "./AppRouteType";
