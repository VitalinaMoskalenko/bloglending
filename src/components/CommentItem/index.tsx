import React from "react";
import styled from "styled-components";
import { Body } from "../Paragraphs";
import { H5, H3 } from "../Headings";
import { BreakpointType } from "../../types";
import FavoriteToggleStories from "../FavoriteToggle/stories/FavoriteToggle.stories";
import { FavoriteToggle } from "..";
// import { Like } from "..";

const Container = styled.li`
  display: flex;
  flex-direction: column;
  background-color: ${({ theme }) => theme.colors.white};
  height: fit-content;
  padding: 18px;
  border-radius: 8px;
  margin-bottom: 10px;
`;

const Name = styled(H3)`
  text-align: center;
  margin-bottom: 10px;
`;

const CommentBottom = styled.div`
  display: flex;
  flex-direction: row-reverse;
  justify-content: space-between;
  align-items: center;
  margin-top: 10px;
`;

const Author = styled(H5)`
  text-align: right;
`;

type PropsType = {
  isToggle: boolean;
  name: string;
  body: string;
  email: string;
  onClick: () => void;
};

const CommentItem = ({ name, body, email, isToggle, onClick }: PropsType) => {
  return (
    <Container>
      <Name>{name}</Name>
      <Body>{body}</Body>
      <CommentBottom>
        <Author>{email}</Author>
        <FavoriteToggle isToggle={isToggle} onClick={onClick} />
      </CommentBottom>
    </Container>
  );
};

export default CommentItem;
