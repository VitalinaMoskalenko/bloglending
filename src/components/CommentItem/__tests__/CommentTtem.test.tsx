import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import CommentItem from "../";
import { theme } from "../../../config";

it("renders commentItem correctly", () => {
  const commentItem = renderer
    .create(
      <ThemeProvider theme={theme}>
        <CommentItem
          name="name"
          body="body"
          email="email"
          isToggle={true}
          onClick={() => {}}
        />
      </ThemeProvider>
    )
    .toJSON();

  expect(commentItem).toMatchSnapshot();
});
