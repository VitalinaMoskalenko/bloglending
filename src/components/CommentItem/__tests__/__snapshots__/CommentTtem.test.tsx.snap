// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders commentItem correctly 1`] = `
<li
  className="sc-iwajpm bhrevb"
>
  <h1
    className="sc-eCApnc sc-cxNHIi TwDaM hksLek"
  >
    name
  </h1>
  <p
    className="sc-bdnxRM jupkGm"
  >
    body
  </p>
  <div
    className="sc-lmgQwP hUAZNt"
  >
    <h1
      className="sc-gKAaRy sc-iJCRrE kXpCcI bDMOXt"
    >
      email
    </h1>
    <div>
      <h1
        className="sc-hKFxyN sc-bCwfaz cEUUoP kaCpzI"
        onClick={[Function]}
      >
        ‚ù£
      </h1>
    </div>
  </div>
</li>
`;
